### PR TITLE
introduced mcurl for parallel connections

### PR DESCRIPTION
**Pull request by maybeec**
Tested on windows git bash with success

### CODE REVIEW

**Comment by maybeec** [comment]
needs some more testing - had some issues with some of the software.

File: scripts/src/main/resources/scripts/functions [hunk 1]
```diff
@@ -567,12 +567,8 @@ function doDownloadInternal() {
     doEcho "Artifact already exists at ${target}\nTo force update please delete the file and run again."
     return 255
   fi
-  local curl_opt=""
-  if [ -n "${quiet}" ]
-  then
-    curl_opt="--silent"
-  fi
-  curl ${curl_opt} -fL "${url}" -o "${tmp_file}"
```
Comment (hohwille) [inline scripts/src/main/resources/scripts/functions:575]: you removed quiet/silent support without a replacement.

File: scripts/src/main/resources/scripts/functions [hunk 2]
```diff
@@ -567,12 +567,8 @@ function doDownloadInternal() {
...
+  SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+  bash $SCRIPT_DIR/mcurl -o "${tmp_file}" "${url}"
```
Comment (hohwille) [inline scripts/src/main/resources/scripts/functions:571]: avoid exposing additional variables (see e.g. #5) and instead use `local` variables.
Further, IMHO the way you determine the script dir is not 100% reliable. Simply use `${DEVON_IDE_HOME}/scripts` instead what will work in all cases.

**Review by hohwille — CHANGES_REQUESTED** [review]
This PR introduces build errors. Besides some shellcheck peanuts, I am deeply concerned about this one:
```
*** Setting up mvn ***
Starting installation of java in version 11.0.13_8 to /home/runner/work/ide/ide/scripts/target/integration-test/test-setup/software/java
Trying to download java-11.0.13_8-linux.tar.gz from https://github.com/adoptium/temurin11-binaries/releases/download/jdk-11.0.13%2B8/OpenJDK11U-jdk_x64_linux_hotspot_11.0.13_8.tar.gz
Download https://github.com/adoptium/temurin11-binaries/releases/download/jdk-11.0.13%2B8/OpenJDK11U-jdk_x64_linux_hotspot_11.0.13_8.tar.gz to /tmp/java-11.0.13_8.QfXnH4 with 2 tasks.


gzip: stdin: not in gzip format
tar: Child returned status 1
tar: Error is not recoverable: exiting now
Failed to run command: tar xf '/home/runner/Downloads/devonfw-ide/java-11.0.13_8-linux.tar.gz' -C '/home/runner/work/ide/ide/scripts/target/integration-test/test-setup/updates/extracted/java'

******** ATTENTION ********
ERROR: Failed to run command tar
We are sorry for the inconvenience. Please check the above errors, resolve them and try again.
Exit code was 2
```

So this download works quite reliable without mcurl but now broke. This says me that we will introduce more problems with mcurl than we solve. So without any further investigation or improvements I vote against going for mcurl now.

**Comment by hohwille** [comment]
How should we proceed here?
Are there benchmarks proving that mcurl will bring a huge benefit to speed our downloads and will there be someone available solving the problems and ensuring that mcurl downloads are reliable?
Or should we close this PR and issue #703?

**Comment by maybeec** [comment]
I would like to keep it like a draft and have some additional look at it once I find back some time if you are ok. Otherwise close it and I can reopen it at any point in time

**Comment by hohwille** [comment]
For house-keeping I will close this PR. Feel free to reopen at any time when there is any progress.

**Pull request closed without merging**